### PR TITLE
Bump bootsnap from 1.6.0 to 1.18.3

### PR DESCRIPTION
**Pull request by orien**
### [bootsnap](https://github.com/Shopify/bootsnap)

**Minor** version upgrade :chart_with_upwards_trend::large_orange_diamond: 1.6.0 → 1.18.3

[_[change-log](https://github.com/Shopify/bootsnap/blob/main/CHANGELOG.md), [source-code](https://github.com/Shopify/bootsnap), [gem-diff](https://my.diffend.io/gems/bootsnap/1.6.0/1.18.3)_]

A change of **248** commits. See the full changes on [the compare page](https://github.com/Shopify/bootsnap/compare/v1.6.0...v1.18.3).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -307,8 +307,8 @@ GEM
     bigdecimal (3.1.6)
     binding_of_caller (0.8.0)
       debug_inspector (>= 0.0.1)
-    bootsnap (1.6.0)
-      msgpack (~> 1.0)
+    bootsnap (1.18.3)
```
Comment (orien) [inline Gemfile.lock:310]: ### [bootsnap](https://github.com/Shopify/bootsnap)

**Minor** version upgrade :chart_with_upwards_trend::large_orange_diamond: 1.6.0 → 1.18.3

[_[change-log](https://github.com/Shopify/bootsnap/blob/main/CHANGELOG.md), [source-code](https://github.com/Shopify/bootsnap), [gem-diff](https://my.diffend.io/gems/bootsnap/1.6.0/1.18.3)_]





<details>
<summary>Commits</summary>

A change of **248** commits. See the full changes on [the compare page](https://github.com/Shopify/bootsnap/compare/v1.6.0...v1.18.3).

These are the first 10 commits:
- (3044e5f) [Fix invalid path value](https://github.com/Shopify/bootsnap/commit/3044e5f8be6bd1e4724531a0b35262d2d822a35e)
- (53ace7f) [Expand paths with realpath in YAML cache](https://github.com/Shopify/bootsnap/commit/53ace7f580954e05bff2bd78734e0037b4e96a4b)
- (f52e16b) [Use GC.verify_compaction_references to make sure C reference…](https://github.com/Shopify/bootsnap/commit/f52e16b643893047981e63989cd5ba3abb315362)
- (b941bad) [Allow to turn caches off through environment variables](https://github.com/Shopify/bootsnap/commit/b941badd945e32e4c19cafbd90f9ae59844bc5dc)
- (2d45c5e) [Deprecate ActiveSupport::Dependencies integration](https://github.com/Shopify/bootsnap/commit/2d45c5e2303a2d27e58086d763e19d42341dd36d)
- (8c7845d) [Merge pull request #344 from Shopify/deprecate-activesupport](https://github.com/Shopify/bootsnap/commit/8c7845d864b4a9edd19c7bd102607b16420c7c86)
- (0a1283d) [Deprecate disable_trace option](https://github.com/Shopify/bootsnap/commit/0a1283d0c7de99e527b81d25b4bea1114038932b)
- (9d6ff0e) [Allow to control the behavior of `require 'bootsnap/setup'` …](https://github.com/Shopify/bootsnap/commit/9d6ff0e94ec304fa3e79360e663a98493f4ce7b9)
- (04dfc1b) [Remove outdated README.jp.md](https://github.com/Shopify/bootsnap/commit/04dfc1b5ff2c5330dd44fe2ba5f018d5df0ae26c)
- (4bf7a66) [Enable ISeq caching on 2.5.4+](https://github.com/Shopify/bootsnap/commit/4bf7a665c7a0cb26c2b61e7d4e06f846684706ed)


</details>

File: config/boot.rb [hunk 2]
```diff
@@ -13,7 +13,6 @@
     cache_dir:            'tmp/bootsnap', # Path to your cache
     development_mode:     rails_env == "development",
     load_path_cache:      true, # optimizes the LOAD_PATH with a cache
-    autoload_paths_cache: true, # optimizes ActiveSupport autoloads with cache
```
Comment (orien) [inline config/boot.rb:16]: Bootsnap no longer supports the classic Rails class autoloader, which this option refers to.

Samson is not using the classic autoloader. Rather, it's using Zeitwerk. So removing this capability (and configuration) is of no consequence.

**Review by grosser — APPROVED** [review]

File: test/test_helper.rb [hunk 3]
```diff
@@ -226,7 +226,7 @@ def stub_const(base, name, value)
 
   def self.only_callbacks_for_plugin(callback)
     line = caller(1..1).first
-    plugin_name = line[/\/plugins\/([^\/]+)/, 1] || raise("not called from a plugin not #{line}")
+    plugin_name = line[/^plugins\/([^\/]+)/, 1] || raise("not called from a plugin not #{line}")
```
Comment (orien) [inline test/test_helper.rb:229]: Adjust the plugin-detecting regex to get the tests passing.

**Comment by orien** [comment]
@grosser The test suite is passing now.

**Review by grosser — APPROVED** [review]
weird that the path changed ... but fix seems good 👍

**Pull request merged**